### PR TITLE
DTS: Intel ioapic/mvic and gpio, uart, rtc...

### PR DESCRIPTION
**Pull request by tbursztyka**
A PR that enables a bit more the use of DTS for Intel SoCs as well as the related drivers such as uart ns16550 and qmsi gpio/rtc/uart ones

Note that only the 18 last patches are new, all the first ones are from PR #6406

### CODE REVIEW

**Comment by tbursztyka** [comment]
@erwango I removed you as a reviewer, it automatically added you because of the patches from PR #6406 which is touching some Kconfig stuff around stm32.

**Review by galak — CHANGES_REQUESTED** [review]
So I see _ioapic_irq_set(irq, vector, flags) - I don’t see any means for priority to get set or passed down to the ioapic.  So I don’t see why we would add priority to the dts for iopaic.

Also, we should split this so mvic is one patch, and all fixes to mvic dts’s are in that patch.  If we have something meaningful for ioapic than that should be its own patch, with changes to .dts etc.

File: drivers/gpio/Kconfig.qmsi [hunk 1]
```diff
@@ -50,12 +50,12 @@ config GPIO_QMSI_0
 
 config GPIO_QMSI_0_NAME
 	string "Driver name"
-	depends on GPIO_QMSI_0
+	depends on GPIO_QMSI_0 && !HAS_DTS
```
Comment (galak) [inline drivers/gpio/Kconfig.qmsi]: Are there any cases for !HAS_DTS?  Any reason to keep these Kconfig symbols?

**Comment by tbursztyka** [comment]
@galak as @andrewboie mentioned here https://github.com/zephyrproject-rtos/zephyr/pull/6406#discussion_r172347803

Check include/arch/x86/arch.h

File: drivers/gpio/Kconfig.qmsi (hunk 1 above)
Comment (tbursztyka) [reply drivers/gpio/Kconfig.qmsi]: Has it's been done in all other Kconfig places, config options there are kept with that dependency. 
Let's remove all these options when we are done porting everything to dts, and sure it works properly.

Comment (galak) [reply drivers/gpio/Kconfig.qmsi]: > Has it's been done in all other Kconfig places, config options there are kept with that dependency.
> Let's remove all these options when we are done porting everything to dts, and sure it works properly.

Its been the case that we've been removing Kconfig options if there aren't any uses for non-DTS.  So if its the case that its all users of GPIO_QMSI_0_NAME have HAS_DTS=y, than we should remove the Kconfig symbol.

**Comment by galak** [comment]
> @galak as @andrewboie mentioned here #6406 (comment)
> 
> Check include/arch/x86/arch.h

Based on include/arch/x86/arch.h - it seems as if mvic doesn't support priority, trying to understand the comments about how this work for ioapic.

**Comment by tbursztyka** [comment]
Even though mvic seems to have hard-coded interrupt priorities, tightly related to the interrupt vectors, the priority is still inserted into the IDT, same core interrupt handling, same gen_idt tool used.

Does that mean mvic is handled the same way as ioapic @andrewboie ? Or is the priority in the IDT silently ignored in mvic case?

**Comment by galak** [comment]
@andrewboie any comment on this:

> Even though mvic seems to have hard-coded interrupt priorities, tightly related to the interrupt vectors, the priority is still inserted into the IDT, same core interrupt handling, same gen_idt tool used.
> 
> Does that mean mvic is handled the same way as ioapic @andrewboie ? Or is the priority in the IDT silently ignored in mvic case?

**Comment by tbursztyka** [comment]
@andrewboie we really need your input here :)

**Comment by nashif** [comment]
interrupt vectors in the mvic are fixed and AFAIK you can't program them.

**Comment by tbursztyka** [comment]
@nashif Sure but to keep things consistent within all i2c drivers and dts files, I needed no do these changes (so instead of a hard-coded value in driver, it's provided by dts from soc's dtsi). Now the issue is just about mvic priority.

**Review by tyrantlizard77 — COMMENTED** [review]
On MVIC, it is basically a reduced feature set IOAPIC. IDT Vector assignments are fixed, they are just the IRQ line number + 32. So any priority value provided is just ignored. The only flexibility the OS has wrt priorities is to select a free vector for the timer interrupt (by setting its IRQ line in a register), and hence the priority of the timer. Everything else is graven in stone.

**Comment by galak** [comment]
> On MVIC, it is basically a reduced feature set IOAPIC. IDT Vector assignments are fixed, they are just the IRQ line number + 32. So any priority value provided is just ignored. The only flexibility the OS has wrt priorities is to select a free vector for the timer interrupt (by setting its IRQ line in a register), and hence the priority of the timer. Everything else is graven in stone.

So all of that means that MVIC doesn't support priority and thus shouldn't be in the device tree for MVIC or MVIC related interrupts.

**Pull request closed without merging**